airflow-core/src/airflow/utils/db_cleanup.py

```diff
@@ -81,6 +82,7 @@ class _TableConfig:
     keep_last: bool = False
     keep_last_filters: Any | None = None
     keep_last_group_by: Any | None = None
+    dependent_tables: list[str] | None = None
```

Comment:
We should add a code comment saying why we made this explicit, otherwise I can easily see someone coming and "fixing" it by making the detection automatic.